### ISSUE TITLE
New createFromFormat for CarbonInterval

### ISSUE THREAD

**Issue by xfudox** [briannesbitt/Carbon#1918]
This would be a proposal, not an issue, but i wasn't sure on where to put it.

The proposal is to implement a `createFromFormat($format, $time)` constructor for `CarbonInterval` as `Carbon` object already has, this to better handle MySQL (and others DBMS) `time` data type or other string representations of time intervals.

It is tedious to use a long

```php
Carboninterval::create(0, 0, 0, 0, 1, 30, 0, 0)
```

or even a

```php
Carboninterval::hours(1)->minutes(30)
```

since i have to split the original string into several variables, and eventyally calculate days, weeks and so on.

It would be more readable and user friendly something like 

```php
Carboninterval::createFromFormat('H:i', '1:30')
```

**Comment by kylekatarnls** [comment]
Hi, this sounds very interesting and pretty simple.

I can already give you a macro that would do exactly that:

```php
CarbonInterval::macro('createFromFormat', static function (string $format, string $interval) {
  return Carbon::createFromFormat('!'.ltrim($format, '!'), $interval, 'UTC')->diffAsCarbonInterval(
    Carbon::parse('1970-01-01 00:00:00', 'UTC')
  );
});

echo Carboninterval::createFromFormat('H:i', '1:30');
```

If you want to propose it as a core method of CarbonInterval, I would release it in next minor release.

**Comment by kylekatarnls** [comment]
Will be released in 2.27.0
